mm2src/lp_ordermatch.rs

```diff
@@ -1613,13 +1613,17 @@ impl MakerOrder {
             TakerAction::Sell => {
                 let taker_price = taker_base_amount / taker_rel_amount;
 
+                // Calculate the resulting base amount using the Maker's price instead of the Taker's.
+                let matched_base_amount = taker_base_amount / &self.price;
```

Comment:
Shouldn't we do the same for the `TakerAction::Buy` case?

Reply:
No, since https://github.com/KomodoPlatform/atomicDEX-API/blob/41170748db961f23aa8a10fffc3de09a2bd81aa9/mm2src/lp_ordermatch.rs#L1604-L1606

We have the following inequality:

`min_rel_amount = min_base_amount * price` where `min_base_amount <= taker_base_amount`
`min_rel_amount <= taker_base_amount * price` where `rel_amount = taker_base_amount * price`
So `min_rel_amount <= rel_amount`

`max_rel_amount = max_base_amount * price` where `taker_base_amount <= available_amount <= max_base_amount`
`max_rel_amount >= taker_base_amount * price` where `rel_amount = taker_base_amount * price`
So `max_rel_amount >= rel_amount`

I hope everything is correct :D

Reply:
I see, thanks for the response :slightly_smiling_face: